media/multipart/src/main/java/org/glassfish/jersey/media/multipart/ContentDisposition.java

```diff
@@ -192,6 +205,42 @@ private void createParameters() throws ParseException {
         size = createLong("size");
     }
 
+    private String defineFileName() throws ParseException {
+        final String fileName = parameters.get("filename");
+
+        final String fileNameExt = parameters.get("filename*");
+        if (fileNameExt == null) {
+            return fileName;
+        }
+
+        final Matcher matcher = FILENAME_EXT_VALUE_PATTERN.matcher(fileNameExt);
+        if (matcher.matches()) {
+            if (isEncodedInUriFormat(fileNameExt)) {
+                return fileNameExt;
```

Comment:
This never happens in your tests, since `isEncodedInUriFormat(fileNameExt)` always fail on `'`. Perhaps you meant `if (isEncodedInUriFormat(matcher.group(FILENAME_GROUP_NAME)))`?

But then again, why you do `if (matcher.matches())` ? If `isEncodedInUriFormat(matcher.group(FILENAME_GROUP_NAME))` is true, we likely should not care about the charset. You are right, the obsolete RFC 5987 requires UTF-8 or ISO-8859-1, but then new RFC 8187 changes the charset to UTF-8, but then next month another RFC supporting UTF-16 can arise - do you see a reason for not supporting it at the moment when it is already encoded by the user?

Reply:
You are right about `if (isEncodedInUriFormat(matcher.group(FILENAME_GROUP_NAME)))` and I fixed it. 

`if (matcher.matches())` uses to check if filename* parameter has correct structure and also it uses to check language value in this parameter.

I agree with you that it is not necessary to require only UTF-8 or ISO-8859-1 charset if filename is encoded, so I made necessary changes.